editor/writing-flow/index.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from 'element';
+import { keycodes } from '@wordpress/utils';
+
+/**
+ * Internal dependencies
+ */
+import { isEdge, placeCaretAtEdge } from '../utils/dom';
+
+/**
+ * Module Constants
+ */
+const { UP, DOWN, LEFT, RIGHT } = keycodes;
+
+class WritingFlow extends Component {
+	constructor() {
+		super( ...arguments );
+		this.zones = [];
+		this.onKeyDown = this.onKeyDown.bind( this );
+		this.onKeyUp = this.onKeyUp.bind( this );
+		this.bindContainer = this.bindContainer.bind( this );
+		this.state = {
+			shouldMove: false,
+		};
+	}
+
+	bindContainer( ref ) {
+		this.container = ref;
+	}
+
+	getVisibleTabbables() {
+		const tabblablesSelector = [
```

Comment:
As in #2323 and #2392, we're increasingly drawn to needing general helpers with this sort of logic, and am tempted to bring in [ally.js](https://allyjs.io/) despite it not being terribly compact. Don't need to make changes here, but curious if you have thoughts.

Reply:
Using a library or extracting to a general helper seems like a good thing to me. But I'm not sure it will work as expected, if I understand correctly `ally.js` returns different tabbables depending on the browser. We'll have to test I guess

Reply:
Should this be `tabbablesSelector`?